feat(issues): Add highlights and remove duplicate data (#10042)
getsentry/sentry-docs: docs/product/issues/issue-details/index.mdx

![Issue details](./img/issue-details-page.png)

You can view information about a specific error event that's part of an issue on the <SandboxLink scenario="oneIssue" projectSlug="react">Issue Details</SandboxLink> page. You can also navigate between the events of an issue using the event navigator.

In the right hand sidebar, [sentry.io](https://sentry.io) reflects a summary that includes information such as how often the <SandboxLink scenario="oneIssue" projectSlug="react">issue</SandboxLink> has occurred in the last 24 hours and the last 30 days, as well as the last time and the first time the issue was seen. If the issue is linked to any GitHub or Jira issues, that's displayed here as well. Lastly, there is a facet map, which shows the distribution of values for the tags for all events included in the issue. All of these values are based on the environment that you've selected in the environment dropdown.

> ## Issue Tabs

Author: Moved this section up to the top of the page since it seems like this page follows the same pattern as the issue details page, top to bottom.

- **All Events** - A list of all the error events included in the issue.
- **Merged Issues** - A list of error issues that have been merged with this one.
- **Similar Issues** - A list of error issues that have a similar stack trace [that you can merge](/concepts/data-management/event-grouping/merging-issues/).
- **Replays** – A list of [replays](/product/session-replay/) tied to error events included in the issue.

## Issue Categories & Page Details

There are two categories of issues: [error issues](/product/issues/issue-details/error-issues/) and [performance issues](/product/issues/issue-details/performance-issues/). Depending on the issue category and available event data, the **Issue Details** page displays some combination of the following sections:

- **Event Navigator** - Allows you to select between the latest, oldest, or a recommended issue event and navigate between events in sequence.
- **Trace Navigator** - An abbreviated view of the related [trace](/product/sentry-basics/tracing/distributed-tracing) for the current transaction.
- **Suspect Commits** - A commit that's been identified as potentially having caused an error event.

> - **Event Highlights** - Specific [tags](#tags) and [context](#contexts) values that are useful for debugging

Reviewer: ```suggestion
- **Event Highlights** - Specific [tags](#tags) and [context](#contexts) values that are useful for debugging.
```

- **Stack Trace** - Shows you the line of code that the event errored on.

> - **Span Evidence** - Information about the performance problem in the context of the current event

Reviewer: ```suggestion
- **Span Evidence** - Information about the performance problem in the context of the current event.
```

- **Breadcrumbs** - Provide a history and timeline leading up to the error event and are customizable.

> - **Tags** - Searchable key/value string pairs providing information such as the browser or device.

Reviewer: ```suggestion
- **Tags** - Searchable key/value string pairs providing information, such as the browser or device.
```

> - **Screenshot** - Screenshot taken when a user experiences an error, an exception or a crash; only available on [certain SDKs](/product/issues/issue-details/error-issues/#screenshots).

Reviewer: ```suggestion
- **Screenshot** - Screenshot taken when a user experiences an error, an exception, or a crash; only available on [certain SDKs](/product/issues/issue-details/error-issues/#screenshots).
```

> - **Contexts** - Non-searchable key/value data that helps provide context (haha) in debugging the issue.

Reviewer: ```suggestion
- **<PlatformLink to="/enriching-events/context/">Context</PlatformLink>** - Non-searchable key/value data that helps provide context (haha) in debugging the issue.
```
I think it's worth including a link like we do in the product so it's clear where this info comes from. Hope the link is correct!

![Trace Navigator](./img/trace-navigator.png)

> The trace navigator (which displays below the date) is an abbreviated view of the related [trace](/product/sentry-basics/tracing/distributed-tracing) for the current transaction. It displays up to six nodes, each representing different groups of the event's trace:

Author: This isn't new documentation. This page had the older copy/screenshots of the Trace Navigator which were taking from `/issus/issue-details/error-issues/` and then removed from there.

> The trace navigator (which displays below the date) is an abbreviated view of the related [trace](/product/sentry-basics/tracing/distributed-tracing) for the current transaction. It displays up to six nodes, each representing different groups of the event's trace:

Reviewer: Are nodes the same as the dots in the timeline? If so, the image shows more than 6 so this is kinda confusing. I'm also not sure how to identify the six nodes in the timeline. For example, how do I tell ancestor vs parent vs children vs descendants apart? Are they all represented the same way?

```suggestion
The trace navigator (which is displayed below the date) is an abbreviated view of the related [trace](/product/sentry-basics/tracing/distributed-tracing) for the current transaction. It displays six types of nodes (if applicable), each representing different groups of the event's trace and visualized as a dot in the timeline:
```

Author: I'll pass this feedback along to the team to follow up on, I don't know too much about the trace navigator.

A _suspect commit_ is a commit that's been identified as potentially having caused the error event. If you've configured [suspect commits](/product/issues/suspect-commits/), these are displayed. This allows you to immediately investigate commits that are likely to have generated the issue. The author of the suspect commit is also suggested as an assignee for the issue.

> ## Event Highlights

Author: This is new copy

![Event Highlights](./img/issue-highlights.png)

> _Event Highlights_ are specific [tags](#tags) and [context](#contexts) items that have been promoted to the top of the page for quicker debugging. If you miss tags being at the top of the page, the 'View All' will send you to their new location under the stack trace.

Reviewer: ```suggestion
_Event Highlights_ are specific [tags](#tags) and [context](#contexts) items that have been promoted to the top of the page for quicker debugging. If you miss tags being at the top of the page, click "View All" to go to their new location under the stack trace.
```

> By default, Sentry will pick out a few common ones based on your project's platform, but these are intended to be customized to best fit your needs. Project Admins can click on the 'Edit' button to add or remove any highlights they see fit and apply those changes across the project.

Reviewer: ```suggestion
By default, Sentry will surface a few common tags and context values based on your project's platform. However, these highlights are intended to be customized to best fit your needs. Project Admins can click on the "Edit" button to add or remove any highlights they see fit and apply those changes across the project.
```

If you’ve enabled [Session Replay](/product/session-replay/), you’ll see a replay preview under Breadcrumbs if there’s one associated with the event you’re viewing. Replays can be associated with both frontend and [backend errors](/product/session-replay/getting-started#replays-for-backend-errors) (as long as distrubted tracing is set up). Clicking on the replay preview will lead you to the [Replay Details](/product/session-replay/replay-details/) page.

> ## Trace Details

Author: Opted to remove this section since it describes one type of context when there are endless others. At best, it links out to distributed tracing (something the #Trace Navigator section already does) and mentions that there's a link.

Instead I've added the contexts section to replace it.

<Include name="common-imgs/tags" />

[Tags](/product/sentry-basics/search/) are key/value string pairs that are both indexed and searchable. For example, a tag provides you with information such as the browser, device, or user associated with the event. Tags are the diagnostic information sent by the SDK for the individual event.

> These event tags can be filtered by one of the available categories, and clicking any of the hover dropdowns will provide you with a few more search options. Sentry will display tags containing '`.`' characters as a tree structure to make things a bit more readable, but searching should always use the full tag name (e.g. `organization.slug` instead of `slug`).

Author: This middle paragraph is new copy as well.

> These event tags can be filtered by one of the available categories, and clicking any of the hover dropdowns will provide you with a few more search options. Sentry will display tags containing '`.`' characters as a tree structure to make things a bit more readable, but searching should always use the full tag name (e.g. `organization.slug` instead of `slug`).

Reviewer: ```suggestion
You can use the tag tabs to filter these event tags by category ("All", "Custom", "Application", "Other").  If you hover over a tag and then click "...", you can choose to view more events or issues with that tag value (and sometimes other information associated with that tag, such as the release itself). Sentry will display tags containing '`.`' characters as a tree structure to make things a bit more readable, but searching should always use the full tag name (for example, `organization.slug` instead of `slug`).
```

The tags displayed in the main section of the page are specific to the event that you're viewing. The tags displayed in the right-hand sidebar are a summary of all tag values for all events included in the issue. You can set your own tags to make them more useful for debugging as described in <PlatformLink to="/enriching-events/tags/">Customize Tags</PlatformLink>.

## Screenshot

> Sentry provides the ability to take a screenshot and include it as an attachment when a user experiences an error, an exception or a crash. Screenshot attachments sent as a part of an event are displayed in this section. This feature only applies to SDKs with a user interface. It's supported for the following SDKs:

Reviewer: ```suggestion
Sentry provides the ability to take a screenshot and include it as an attachment when a user experiences an error, an exception, or a crash. Screenshot attachments sent as part of an event are displayed in this section. This feature only applies to SDKs with a user interface. It's supported for the following SDKs:
```

- [.NET Xamarin](/platforms/dotnet/guides/xamarin/)
- [Android](/platforms/android/enriching-events/screenshots/)
- [Flutter](/platforms/flutter/enriching-events/screenshots/)
- [iOS](/platforms/apple/guides/ios/enriching-events/screenshots/)
- [JavaScript Electron](/platforms/javascript/guides/electron/enriching-events/screenshots/)
- [React Native](/platforms/react-native/enriching-events/screenshots/)
- [Unity](/platforms/unity/enriching-events/screenshots/)

> ## Contexts

Author: This is all new copy

> ## Contexts

Reviewer: ```suggestion
## Context
```
Since it isn't plural in the UI

Author: I'm actually changing this in an open PR, since `Contexts` is more accurate, but great catch!

![Context Section](./img/issue-context-section.png)

> Contexts contain related key/value items (not just strings) that you've set manually in your application, or have been automatically added when Sentry processes an event. Contexts are meant to provide you with related sets of data to help with your debugging.

Reviewer: ```suggestion
Context contains related key/value items (not just strings) that you've set manually in your application or have been automatically added by Sentry. Contexts are meant to provide you with related sets of data to help with your debugging.
```

Author: Done manually with updating screenshots and other references to 'Contexts'

> The issue details page only displays context for the individual event, and context cannot be searched. If you require searching for your data, refer to [Tags](#tags)

Reviewer: ```suggestion
The issue details page only displays context for the individual event, and context cannot be searched. If you require searching for your data, use [Tags](#tags) instead.
```

> You can setup your own context items to collect useful debugging infomation in your next event, as described in <PlatformLink to="/enriching-events/context/">Structured Context</PlatformLink>.

Reviewer: ```suggestion
You can set up your own context items to collect useful debugging information in your next event, as described in <PlatformLink to="/enriching-events/context/">Structured Context</PlatformLink>.
```
